src/helpers/host-component-names.tsx

```diff
@@ -4,9 +4,12 @@ import TestRenderer from 'react-test-renderer';
 import { HostComponentNames } from '../config';
 import { getQueriesForElement } from '../within';
 
+const defaultErrorMessage =
+  'There seems to be an issue with your configuration that prevents the library from working correctly. Please ensure that you have a version of react native compatible with the library';
+
 export function detectHostComponentNames():
-  | HostComponentNames
-  | { text: null; textInput: null } {
+  | (HostComponentNames & { errorMessage?: never })
+  | { errorMessage: string; text?: never; textInput?: never } {
```

Comment:
If we want to signal an error while returning error message, the idiomatic JS is throwing error with proper message. 

@pierrezimmermannbam You correctly indicated that thrown errors are not declared in return type (unlike e.g. Swift and Java) but they still are idiomatic way of implementing error handling in JS, and are generally understood by JS devs. Proposed solution is handling error in a type safe way but the tradeoff here is that we are using quite a weird day that decreases readability of the code. 

Since we know our codebase, and we have a wide test coverage that would prevent us from forgetting to handle a thrown error, I think that a better tradeoff is to go with throwing errors.